raster/r.carve/enforce_ds.c

```diff
@@ -31,179 +33,308 @@
+                          const RASTER_MAP_TYPE rtype);
 static double lowest_cell_near_point(void *data, const RASTER_MAP_TYPE rtype,
-				     const double px, const double py,
-				     const double rad);
+                                     const double px, const double py,
+                                     const double rad);
 static void process_line_segment(const int npts, void *rbuf, Point2 * pgxypts,
-				 Point2 * pgpts, struct BM *bm,
-				 struct Map_info *outMap,
-				 const struct parms *parm);
-
-
-/******************************************************************
- * Returns 0 on success, -1 on error, 1 on warning, writing message
- * to errbuf for -1 or 1 */
-
-int enforce_downstream(int infd, int outfd,
-		       struct Map_info *Map, struct Map_info *outMap,
-		       struct parms *parm)
+                                 Point2 * pgpts, struct BM *bm,
+                                 struct Map_info *outMap,
+                                 const struct parms *parm);
+double get_value(unsigned short int *ctype, dbColumn *col);
+void set_value(dbColumn *col, unsigned short int *ctype, char *answer,
+               double *parm, double *def_value, dbTable *table,
+               struct Cell_head *wind, value_type type);
+struct sql_statement create_select_sql_statement(struct Map_info *Map,
+                                                 struct field_info *Fi,
+                                                 struct boxlist *box_list,
+                                                 char *columns[2],
+                                                 unsigned short int *field,
+                                                 char *keycol);
+
+
+void enforce_downstream(int infd, int outfd,
+                       struct Map_info *Map, struct Map_info *outMap,
+                       struct parms *parm, struct field_info *Fi,
+                       int *width_col_pos, int *depth_col_pos,
+                       char *columns[2], dbDriver *driver)
+/*
+ * Function: enforce_downstream
+ * -------------------------
+ * Enforce downstream
+ *
+ * infd: input raster elevation map
+ * outfd: output raster elevation map with carved streams
+ * Map: input streams vector map
+ * outMap: output vector map for adjusted stream points
+ * parm: module parameters
+ * Fi: layer (old: field) information
+ * width_col_pos: width column position in the columns array
+ * depth_col_pos: depth column position in the columns array
+ * columns: array of width and depth column names
+ * driver: db driver
+ *
+ */
 {
     struct Cell_head wind;
-    int retval = 0;
-    int line, nlines;
     void *rbuf = NULL;
-
-    /* width used here is actually distance to center of stream */
-    parm->swidth /= 2;
+    struct bound_box box;
+    struct boxlist *box_list;
+    int more;
+    unsigned int c;
+    unsigned short int field, width_col_type, depth_col_type;
+    // Width used here is actually distance to center of stream
```

Comment:
```suggestion
    /* Width used here is actually distance to center of stream */
```
(as per https://trac.osgeo.org/grass/wiki/Submitting/C#Comments please avoid C++-style comments in C as being not portable)

Reply:
Thanks for the warning, I didn't notice. I've already fixed it.